src/utils.ts

```diff
@@ -33,9 +33,29 @@ async function normalizeStore(source: string | ZarrArray['store']) {
 
 export async function open(source: string | ZarrArray['store']) {
   const { store, path } = await normalizeStore(source);
-  return openGroup(store, path).catch((err) => {
+
+  function nested(store) {
+    const get = (target, key) => {
```

Comment:
Typescript is super flexible (almost too flexible) when using proxies, but I think all you need here is:

```typescript
  function nested(store: ZarrArray['store']) {
    const get = (target: ZarrArray['store'], key: unknown) => {
```